### PR TITLE
Improve packaging

### PR DESCRIPTION
**Pull request by awf**
This PR adds a `pyproject.toml` file which makes `pyscf_ipu` an installable package.  In particular, it means that we don't explicitly edit sys.path in imported files such as nanoDFT.  It requires a one-time `pip install -e .`, but does not otherwise interfere with development.

Testing:
```
python density_functional_theory.py -methane -backend cpu
python nanoDFT/nanoDFT.py 
jupyter nbconvert --to python notebooks/nanoDFT-demo.ipynb ; python notebooks/nanoDFT-demo.py
```

### CODE REVIEW

File: exchange_correlation/b3lyp.py [hunk 1]
```diff
@@ -2,11 +2,8 @@
 import jax.numpy as jnp
 import jax
 
-import sys
-import pathlib
-sys.path.append( str(pathlib.Path().resolve()) + "/" )
-
 try:
+    # When do we expect these to fail?
```
Comment (AlexanderMath) [inline exchange_correlation/b3lyp.py:6]: If user runs `python b3lyp.py` from directory `exchange_correlation/`. This previously ran test in __main__ (this no longer exists). Feel free to remove `try` and the `except` block.

File: nanoDFT/experimental_pmap_nanoDFT.py [hunk 2]
```diff
@@ -2,14 +2,8 @@
 import jax
 import jax.numpy as jnp
 import numpy as np
-import os
-os.environ['OMP_NUM_THREADS'] = "16"
```
Comment (AlexanderMath) [inline nanoDFT/experimental_pmap_nanoDFT.py:6]: This was used when running multiple IPUs in parallel by starting different python processes. So we definitely want to remove here, nice catch!

File: nanoDFT/experimental_pmap_nanoDFT.py [hunk 3]
```diff
@@ -2,14 +2,8 @@
...
 import pyscf
 from jsonargparse import CLI, Namespace
-import sys
-sys.path.append("../")
-import os 
-os.environ['TF_POPLAR_FLAGS'] = """--executable_cache_path=/tmp/pyscf-ipu-cache/"""
```
Comment (AlexanderMath) [inline nanoDFT/experimental_pmap_nanoDFT.py:12]: I'm on the fence as to whether we want to cache by default. One solution: add a `--cache` option and discuss whether it should be true/false by default.

Comment (awf) [reply nanoDFT/experimental_pmap_nanoDFT.py:12]: See https://github.com/graphcore-research/pyscf-ipu/pull/30

**Pull request merged**